test/test_signal_wait.py

```diff
@@ -230,6 +230,142 @@ def gmem_multi_bar_sync_kernel(signal_pad: torch.Tensor) -> torch.Tensor:
         )
         self.assertIn("atomic_cas", code)
 
+    def test_wait_pointers(self):
+        @helion.kernel
+        def gmem_wait_pointers_kernel(
+            signal_pad_ptrs: torch.Tensor, pad_shape: hl.constexpr
+        ) -> torch.Tensor:
+            out = torch.empty(
+                pad_shape, device=signal_pad_ptrs.device, dtype=torch.int32
+            )
+            N = signal_pad_ptrs.size(0)
+            for i in hl.grid(pad_shape):
+                offset = i * 4  # number of btypes of each pointer (torch.int32)
```

Comment:
Calculating offset to pointers is kinda ugly here and it breaks the no-index calculation / pointer-handling abstraction. The tensor metadata (i.e. size, stride etc.) is lost when passed in as Tensor of data_ptrs, therefore offset calculation need to be explicit here for indirect barriers.

Reply:
Should we pass the offsets into the function so we can take into account the size of the dtype?